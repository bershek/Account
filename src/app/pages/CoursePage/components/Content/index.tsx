import React, { useCallback, useMemo, useRef, useState } from 'react';
import { Button } from 'app/components/Button';
import { ILecture } from '@utils/coursesData';
import { useCoursesSlice } from '../../slice';
import { useDispatch, useSelector } from 'react-redux';
import { getPassedCourses } from '../../slice/selectors';
import Table from '../Table';
import { Wrapper } from 'app/components/Wrapper';
import styled from 'styled-components/macro';

interface ContentProps {
  id: string,
  handleBack(): void
  description: string
  content: Array<ILecture>
}

type CallbackType = (...args: {name: string, value: string}[]) => void

const Content = ({ id, description, content }: ContentProps): JSX.Element => {
  const [searchValue, setSearchValue] = useState<string>('');
  const searchRef = useRef<HTMLInputElement>(null);
  const handleSearchValue = useCallback(() =>
    setSearchValue(searchRef?.current?.value || ""), []);
  const { actions } = useCoursesSlice();
  const dispatch = useDispatch();
  const passedCoursesState = useSelector(getPassedCourses);
  const handlePassLectures = useCallback<CallbackType>((data) =>
    dispatch(actions.passedLectures(data)), []);
  const handleSetComment = useCallback<CallbackType>((data) =>
  {dispatch(actions.addComment(data))}, []);
  const passedCourses = useMemo(() => (passedCoursesState[id] || []), [passedCoursesState, id]);

  const tableContent = useMemo(() =>
      searchValue
        ? content.filter(({title}) => {
          const value = title.toLocaleLowerCase();
          const checkValue = searchValue.trim().toLocaleLowerCase();
          return value.includes(checkValue);
        })
        : content,
    [searchValue]);
  console.log("tableContent", tableContent);
  return (
    <Wrapper>
      <Title>{description}</Title>
      <SearchWrap>
        <input type="text" ref={searchRef} placeholder="Type lecture name"/>
        <Button label="Search" onClick={handleSearchValue}/>
      </SearchWrap>
      <Table
        title={id}
        content={tableContent}
        searchValue={searchValue}
        handlePassLectures={handlePassLectures}
        handleSetComment={handleSetComment}
        passedCourses={passedCourses}
      />
    </Wrapper>
  );
};

const SearchWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  column-gap: 1.25rem;
  margin: 2rem 0;
`;

const Title = styled.h1`
  font-size: 1.5rem;
  font-weight: bold;
  color: ${p => p.theme.text};
  margin: 1rem 0;
`;

export default Content;
