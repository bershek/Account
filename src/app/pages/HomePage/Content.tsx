import React from 'react';
import styled from 'styled-components/macro';
import CourseList from "./CourseList";

const Content = (): JSX.Element => {
  return (
    <Wrapper>
      <Title>Courses</Title>
      <CourseList />
    </Wrapper>
  );
};

const Wrapper = styled.main`
  height: 60vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 320px;
`;

const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: ${p => p.theme.text};
  margin: 1rem 0;
`;


export default Content;
