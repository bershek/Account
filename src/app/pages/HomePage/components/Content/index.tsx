import React from 'react';
import styled from 'styled-components/macro';
import { CourseList } from '../CourseList';
import { Wrapper } from 'app/components/Wrapper';

const Content = (): JSX.Element => {
  return (
    <Wrapper>
      <Title>Courses</Title>
      <CourseList />
    </Wrapper>
  );
};

const Title = styled.h1`
  font-size: 3rem;
  font-weight: bold;
  color: ${p => p.theme.text};
  margin: 1rem 0;
`;

export default Content;
