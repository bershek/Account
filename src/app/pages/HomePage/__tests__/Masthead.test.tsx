import * as React from 'react';
import Content from '../Content';
import { createRenderer } from 'react-test-renderer/shallow';

const shallowRenderer = createRenderer();

describe('<Content />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(<Content />);
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
