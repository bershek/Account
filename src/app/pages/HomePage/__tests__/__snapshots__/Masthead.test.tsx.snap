// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render and match the snapshot 1`] = `
<Content__Wrapper>
  <section
    className="udlite-container"
  >
    <h2
      className="udlite-heading-xl top-categories--title--261i0"
    >
      Courses
    </h2>
    <ul>
      <li>
        <a
          href="/courses/design/"
        >
          <span>
            Design
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/development/"
        >
          <span>
            Development
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/marketing/"
        >
          <span>
            Marketing
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/it-and-software/"
        >
          <span>
            IT and Software
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/personal-development/"
        >
          <span>
            Personal Development
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/business/"
        >
          <span>
            Business
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/photography/"
        >
          <span>
            Photography
          </span>
        </a>
      </li>
      <li>
        <a
          href="/courses/music/"
        >
          <span>
            Music
          </span>
        </a>
      </li>
    </ul>
  </section>
</Content__Wrapper>
`;
